README.md

```diff
@@ -1 +1,127 @@
 # cairo-coverage
+
+## Example
+
+Let's say you have a following program
+
+```rust
+pub enum Operation {
+    Add,
+    Multiply,
+}
+
+
+fn add(a: i32, b: i32) -> i32 {
+    a + b
+}
+
+fn multiply(a: i32, b: i32) -> i32 {
+    a * b
+}
+
+pub fn calculator(a: i32, b: i32, operation: Operation) -> i32 {
+    match operation {
+        Operation::Add => add(a, b),
+        Operation::Multiply => multiply(a, b),
+    }
+}
+
+```
+
+and you cover it with tests
+
+```rust
+#[test]
+fn calculator_add() {
+    assert_eq!(calculator(2, 3, Operation::Add), 5);
+    assert_eq!(calculator(-1, 1, Operation::Add), 0);
+}
+```
+
+When running with `cairo-coverage` you will get a coverage report in `.lcov` format:
+
+```lcov
+TN:
+SF:/path/to/your/project/src/lib.cairo
+FN:7,8,add
+FN:11,12,multiply
+FN:15,20,calculator
+FNDA:2,add
+FNDA:0,multiply
+FNDA:2,calculator
+FNF:3
+FNH:2
+DA:7,2
+DA:8,2
+DA:11,0
+DA:12,0
+DA:15,2
+DA:16,2
+DA:17,2
+DA:18,0
+DA:19,2
+LF:9
+LH:6
+end_of_record
+```
+
+Let's break it down
+
+### Explanation
+
+1. **General Information**
+    - **TN**: Test Name (optional, left empty)
+    - **SF**: Source File path `/path/to/your/project/src/lib.rs`
+
+2. **Function Summary**
+    - **FNF:3**: The number of functions found in the source file. There are 3 functions: `add`, `multiply`,
+      and `calculator`.
+    - **FNH:2**: The number of functions that were hit. 2 out of the 3 functions were
+      executed: `add` and `calculator`.
+
+3. **Function Details**
+    - **FN:7,8,add**: The `add` function starts at line 7 and ends at line 8.
+    - **FN:11,12,multiply**: The `multiply` function starts at line 11 and ends at line 12.
+    - **FN:15,20,calculator**: The `calculator` function starts at line 15 and ends at line 20.
```

Comment:
We do not have this information 🤔. What is the purpose of it?

Reply:
This information is [here](https://github.com/software-mansion/cairo-coverage/blob/80b77f105ce4af0b4d64cc8bf380f8a2fc6284dd/README.md?plain=1#L48). The output i wrote is inspired by real run on C code, using [this](https://github.com/linux-test-project/lcov/blob/d465f73117ac3b66e9f6d172346ae18fcfaf0f69/README#L194) guide.

Reply:
Sorry I think you misunderstood my question 😅 (As i read it it is absolutely not obvious what I meant). I meant we cannot extract this information from the existing mappings, what is the puprose of this information in lcov and how we can work it around.

Reply:
If we take into consideration that we don't have information on where functions start and end, we encounter a problem. The `genhtml` tool, which generates HTML visualizations from `lcov`, will throw an error if it doesn't have `FN`  information.
```
Reading tracefile app.info.
genhtml: ERROR: (corrupt) unable to read trace file 'app.info': genhtml: ERROR: (mismatch) unknown function 'add'
```

You can still run it with `genhtml --ignore-errors mismatch` and produce visualization, but you will always get 100% function coverage even if it is not accurate.

Reply:
By maybe there is a way to make behave properly or we will have to look for new tools